src/posix/platform/udp.cpp

```diff
@@ -316,11 +348,27 @@ otError otPlatUdpConnect(otUdpSocket *aUdpSocket)
 #ifdef __APPLE__
         sin6.sin6_family = AF_UNSPEC;
 #else
+        char      netifName[IFNAMSIZ];
+        socklen_t len = sizeof(netifName);
+
+        if (getsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, &netifName, &len) != 0)
+        {
+            otLogWarnPlat("Failed to read socket bound device: %s", strerror(errno));
+            len = 0;
+        }
+
         // There is a bug in linux that connecting to AF_UNSPEC does not disconnect.
         // We create new socket to disconnect.
         SuccessOrExit(error = otPlatUdpClose(aUdpSocket));
         SuccessOrExit(error = otPlatUdpSocket(aUdpSocket));
         SuccessOrExit(error = otPlatUdpBind(aUdpSocket));
+
+        if (len > 0 && netifName[0] != '\0')
+        {
+            fd = FdFromHandle(aUdpSocket->mHandle);
+            VerifyOrExit(setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, &netifName, len) == 0, error = OT_ERROR_FAILED);
+        }
```

Comment:
put before `SuccessOrExit(error = otPlatUdpBind(aUdpSocket));`

Reply:
What's the difference?